client/packages/common/src/ui/components/inputs/TextInput/NumericTextInput.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { FC } from 'react';
+import { StandardTextFieldProps } from '@mui/material';
+import { BasicTextInput } from './BasicTextInput';
+import { NumUtils } from '@common/utils';
+export interface NumericTextInputProps
+  extends Omit<StandardTextFieldProps, 'onChange'> {
+  onChange?: (value: number | undefined) => void;
+  width?: number | string;
+  defaultValue?: number;
+  allowNegative?: boolean;
+  min?: number;
+  max?: number;
+  precision?: number;
+}
+
+export const NumericTextInput: FC<NumericTextInputProps> = React.forwardRef(
+  (
+    {
+      sx,
+      InputProps,
+      width = 75,
+      onChange,
+      defaultValue,
+      allowNegative,
+      min = allowNegative ? -Infinity : 0,
```

Comment:
Rather than infinity, it would be safer to use `NumUtils.MAX_SAFE_API_INTEGER`
The infinite value, or the max value for an integer would both cause an api exception to be thrown